app/commands/api/v1/reports/fetch_keyword.rb

```diff
@@ -0,0 +1,187 @@
+module Api
+  module V1
+    module Reports
+      class FetchKeyword
+        prepend SimpleCommand
+        prepend SimpleResponse
+
+        def initialize(keyword_id)
+          @keyword_id = keyword_id
+        end
+
+        def call
+          return unless keyword
+          print("========>keyword.value : ", keyword.value)
+
+          begin
+            fetch
+          rescue Exception => e
+            print("========>e : ", e)
+            mark_keyword_as_failed
+          ensure
+            update_report
+          end
+        end
+
+        private
+
+        def fetch
+          html_string = get_page_source
+          # html_string = File.open(Rails.root.join('spec', 'mocks', 'crawl_api_google.html'), 'r').read
+          doc = Nokogiri::HTML(html_string)
+
+          if doc.css('#captcha-form').length >= 1
+            raise 'captcha'
+          else
+            total_results, search_time = get_total_results_and_search_time(doc)
+            ad_words_count = get_ad_words_count(doc)
+            links_count = get_links_count(doc)
+
+            keyword.update(
+              ad_words_count: ad_words_count,
+              links_count: links_count,
+              total_results: total_results,
+              search_time: search_time,
+              html_string: html_string,
+              status: Keyword::SUCCESS
+            )
+          end
+        end
+
+        def mark_keyword_as_failed
```

Comment:
This kind of method could be in the model:

```
keyword.mark_as_failed!
```

If we consider `Keyword` to be an Entity (term stemming from [DDD](https://en.wikipedia.org/wiki/Domain-driven_design)), then the entity can define specific behaviors 💭